src/plugins/noBlockedUsers/index.ts

```diff
@@ -0,0 +1,135 @@
+*/
+
+import { definePluginSettings, migratePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import { Logger } from "@utils/Logger";
+import definePlugin, { OptionType } from "@utils/types";
+import { findByPropsLazy } from "@webpack";
+
+const RelationshipStore = findByPropsLazy("getRelationships", "isBlocked");
+
+const settings = definePluginSettings(
+    {
+        hideBlockedMessages: {
+            description: "Hide messages from blocked users",
+            type: OptionType.BOOLEAN,
+            default: true,
+        },
+        ignoreBlockedMessages: {
+            description: "Ignore incoming gateway messages from blocked users (locally)",
+            type: OptionType.BOOLEAN,
+            default: true,
+        },
+        ignoreTyping: {
+            description: "Hide blocked users from the currently typing list in chat",
+            type: OptionType.BOOLEAN,
+            default: true,
+            restartNeeded: true,
+        },
+        hideReferencedAuthor: {
+            description: "Hide blocked authors of referenced messages in replies",
+            type: OptionType.BOOLEAN,
+            default: true,
+            restartNeeded: true,
+        },
+        hideFromMemberList: {
+            description: "Hide blocked users from the members list",
+            type: OptionType.BOOLEAN,
+            default: true,
+            restartNeeded: true,
+        },
+    },
+);
+
+migratePluginSettings("NoBlockedUsers", "NoBlockedMessages");
+
+export default definePlugin({
+    name: "NoBlockedUsers",
+    description: "Hides blocked users and their messages from everywhere possible",
+    authors: [Devs.rushii, Devs.Samu],
+    settings,
+    patches: [
+        {
+            // Hide blocked message groups from non-DM channels
+            find: ".MESSAGE_GROUP_BLOCKED||",
+            replacement: {
+                match: /(\i)\.type===(?:\i\.)+MESSAGE_GROUP_SPAMMER\)\{/,
+                replace: "$& if($1.type === 'MESSAGE_GROUP_BLOCKED' && $self.settings.store.hideBlockedMessages) return;",
+            },
+        },
+        ...[
+            // Ignore new messages from blocked users
+            '"MessageStore"',
+            // Don't mark channels unread because of blocked user messages
+            '"ReadStateStore"',
+        ].map(find => ({
+            find,
+            replacement: {
+                match: /(?<=MESSAGE_CREATE:function\((\i)\){)/,
+                replace: "if($self.settings.store.ignoreBlockedMessages && $self.isBlocked($1.message.author.id)) return;",
+            },
+        })),
+        {
+            // Hides the author of referenced messages in replies to them
+            find: "isRepliedMessage:!!",
+            predicate: () => settings.store.hideReferencedAuthor,
+            replacement: {
+                match: /return null!=.+?SYSTEM_TAG/,
+                replace: "if($self.isBlocked(arguments[0].message.author.id)) return; $&",
```

Comment:
these accesses are dangerous. you should either use optional chaining or destructure them inside your own function with try catch
```suggestion
                replace: "if($self.isBlocked(arguments[0]?.message?.author?.id)) return; $&",
```